pkg/exporter/exporter_test.go

```diff
@@ -214,9 +243,28 @@ func Test_rateLimitExport(t *testing.T) {
 					}})
 			}
 
-			reportInterval := 100 * time.Millisecond
-			// wait for ~2 report intervals to make sure we get a rate-limit-info event
-			time.Sleep(2 * reportInterval)
+			// Wait until we receive the expected number of events and rate-limit-info messages.
+			// Use polling instead of fixed sleep to avoid timing-dependent race conditions.
+			timeout := time.After(500 * time.Millisecond)
+			ticker := time.NewTicker(10 * time.Millisecond)
+			defer ticker.Stop()
+
+		pollLoop:
+			for {
+				gotEvents, gotRateLimitInfo, _ := countEvents(results.items)
+				if gotEvents >= tt.wantEvents && gotRateLimitInfo >= tt.wantRateLimitInfo {
+					// We have all the expected output, proceed to cleanup and assertions.
+					break pollLoop
```

Comment:
I don't think you need to name it in your case?
```suggestion
			for {
				gotEvents, gotRateLimitInfo, _ := countEvents(results.items)
				if gotEvents >= tt.wantEvents && gotRateLimitInfo >= tt.wantRateLimitInfo {
					// We have all the expected output, proceed to cleanup and assertions.
					break
```